### DISCUSSION TITLE
How to top-align text in table with figures?

### DISCUSSION THREAD

**Discussion by michelk** [quarto-dev/quarto-cli#9521] [Q&A]
### Description

When there is a figure in a table (see [documentation](https://quarto.org/docs/authoring/tables.html#html-tables)), how to top-align the  text in the row in the pdf-output?

![Screenshot from 2024-04-29 08-53-42](https://github.com/quarto-dev/quarto-cli/assets/1404919/3b7da85f-f3c3-4e60-ba8f-7b1c0923e7e5)

**Comment by mcanouil** [comment]
Could you share a small **self-contained "working" (reproducible)** example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Reply by cderv** [reply]
@mcanouil the working example is the one from the documentation page linked above - https://quarto.org/docs/authoring/tables.html#html-tables

For PDF, the text is bottom-aligned by default. Question is about how to make it top aligned when a raw HTML table is provided and then parsed by quarto.

**Reply by michelk** [reply]
Thanks for clarification.

**Comment by cderv** [comment]
@michelk I don't think this is easy to vertical-align content provided like this as a Raw HTML table in .qmd

I do think it happens because in LaTeX inserting an image in a table (I believe a `longtable` is used here) will consider the bottom of the image to be the base line and not the top, so text of second cell is aligned with the bottom of the image. 

Not using  an image will correctly place the second cell text vertical align to the top. 

Any control of this would require some LaTeX customization for the `includegraphics` inside the table, and you can't provide using raw HTML table. (for example, [adjustbox](https://www.ctan.org/pkg/adjustbox) CTAN package could help maybe - but requires raw LaTeX input)

So this is a bit of an edge case where using another input table would help here I believe. 

Unless you can replace `\includegraphics` command behavior for just this table environment. 

Anyhow, this is a  LaTeX customization problem

**Reply by michelk** [reply]
As stated [here](https://tex.stackexchange.com/questions/19080/how-to-vertically-center-text-with-an-image-in-the-same-row-of-a-table), we would have to use `\usepackage{graphbox}`. And then pass `align=t` to `\includegraphics`.

But this seems not supported (#6239). Has there been done anything about it? 


Has there been done anything about it?

**Reply by cderv** [reply]
Not yet. Issue is still open. But there will always be a difference between figure we emit and figure emitted by Pandoc. 
Non cross-referencable figure like here I believe, won't be able to be tweaked without Pandoc supporting options. 

This is larger problem, and probably one that we won't tackle very soon. If you need control over LaTeX output, you should emit raw latex directly for now if no easier options exists.
